File: java/src/sml/OutInstruction.java
package sml;

public class OutInstruction extends Instruction{
    private int register;

    public OutInstruction(String label, String opcode) {
        super(label, opcode);
    }

    public OutInstruction(String label, int register){
        this(label, "out");
        this.register = register;
    }

    @Override
    public void execute(Machine m) {
        int value1 = m.getRegisters().getRegister(register);
        System.out.println("R" + register + ": " + value1);
    }

    @Override
    public String toString() {
      //  return super.toString() + result;
        return super.toString();
    }
}

File: kotlin/src/sml/instructions/OutInstruction.kt
package sml.instructions

import sml.Instruction
import sml.Machine

class OutInstruction(label: String, val register: Int) : Instruction(label, "out") {

    override fun execute(m: Machine) {
        println("R$register: ${m.registers.getRegister(register)}")
    }

    override fun toString(): String {
        return super.toString() + " "
    }
}